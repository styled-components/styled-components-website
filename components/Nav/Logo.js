import styled, { css } from 'styled-components';

import rem from '../../utils/rem';

const Logo = styled.div`
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;

  background-position: center;
  background-size: contain;

  ${p =>
    p.compact
      ? css`
          background-image: url(/icon.png);
          width: ${rem(100)};
          height: ${rem(30)};
        `
      : css`
          background-image: url(/nav-logo.png);
          width: ${rem(164)};
          height: ${rem(40)};
        `};
`;

export default Logo;
