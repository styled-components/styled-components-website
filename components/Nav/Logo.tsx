import styled from 'styled-components';
import rem from '../../utils/rem';

const Logo = styled.div`
  background-image: url(/nav-logo.png);
  background-position: center;
  background-size: contain;
  box-sizing: border-box;
  display: inline-block;
  height: ${rem(40)};
  width: ${rem(164)};
  vertical-align: middle;
`;

export default Logo;
