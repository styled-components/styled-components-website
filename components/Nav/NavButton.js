import styled from 'styled-components';

import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import { resetInput } from '../../utils/form';

const NavButton = styled.button`
  ${resetInput};
  background: ${p => (p.active ? 'rgba(0, 0, 0, 0.07)' : 'none')};
  cursor: pointer;
  flex: 0 0 auto;
  height: ${rem(navbarHeight)};
  padding: 0 ${rem(10)};
  text-align: center;
  vertical-align: middle;

  & + & {
    padding-left: 0;
  }
`;

export default NavButton;
