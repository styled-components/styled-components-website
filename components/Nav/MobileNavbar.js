import React from 'react';
import styled, { css } from 'styled-components';
import { Search, KeyboardArrowDown } from '@styled-icons/material';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import { paleGrey } from '../../utils/colors';
import { mobile } from '../../utils/media';
import { CloseIcon, FoldIcon } from './NavIcons';
import Link from '../Link';
import NavLinks from './NavLinks';
import Social from './Social';
import Logo from './Logo';
import NavButton from './NavButton';

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: ${rem(navbarHeight)};
  `)};
`;

const SecondaryMenu = styled.div`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p =>
    p.isOpen
      ? css`
          height: ${rem(navbarHeight)};
        `
      : css`
          height: 0;
        `} display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`;

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
  'aria-label': 'styled components',
}))`
  display: inline-block;
  vertical-align: center;
`;

const ArrowWrapper = styled.div`
  transition: transform 0.2s;

  ${p =>
    p.shouldRotate &&
    css`
      transform-origin: center center;
      transform: rotate(180deg);
    `};
`;

const SecondaryMenuItem = styled.div`
  padding-right: ${rem(20)};
`;

const StyledIcon = styled.div`
  && {
    width: ${p => rem(p.size || 20)};
    height: ${p => rem(p.size || 20)};
  }
`;

const MobileNavbar = props => {
  const { isSideFolded, isMobileNavFolded, onSideToggle, onMobileNavToggle, showSideNav, onSearchButtonClick } = props;

  return (
    <Wrapper>
      {showSideNav !== false && (
        <NavButton active={!isSideFolded} onClick={onSideToggle}>
          {isSideFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>
      )}

      <LogoLink>
        <Logo compact />
      </LogoLink>
      <div>
        <NavButton onClick={onSearchButtonClick}>
          <StyledIcon as={Search} size={28} />
        </NavButton>

        <NavButton onClick={onMobileNavToggle} active={!isMobileNavFolded}>
          <ArrowWrapper shouldRotate={!isMobileNavFolded}>
            <StyledIcon as={KeyboardArrowDown} size={36} />
          </ArrowWrapper>
        </NavButton>
      </div>

      <SecondaryMenu isOpen={!isMobileNavFolded}>
        <NavLinks />
        <SecondaryMenuItem>
          <Social />
        </SecondaryMenuItem>
      </SecondaryMenu>
    </Wrapper>
  );
};

export default MobileNavbar;
