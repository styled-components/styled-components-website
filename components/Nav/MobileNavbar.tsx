import { KeyboardArrowDown } from '@styled-icons/material';
import styled, { css } from 'styled-components';
import { paleGrey } from '../../utils/colors';
import { mobile } from '../../utils/media';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import Link from '../Link';
import Logo from './Logo';
import NavButton from './NavButton';
import { CloseIcon, FoldIcon } from './NavIcons';
import NavLinks from './NavLinks';
import Social from './Social';

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    align-items: center;
    display: flex;
    flex-grow: 1;
    height: ${rem(navbarHeight)};
    justify-content: space-between;
  `)};
`;

const SecondaryMenu = styled.div<{ $isOpen?: boolean }>`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p =>
    p.$isOpen
      ? css`
          height: ${rem(navbarHeight)};
        `
      : css`
          height: 0;
        `} display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`;

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
  'aria-label': 'styled components',
}))`
  display: inline-block;
  vertical-align: center;
`;

const ArrowWrapper = styled.div<{ $shouldRotate?: boolean }>`
  transition: transform 0.2s;

  ${p =>
    p.$shouldRotate &&
    css`
      transform-origin: center center;
      transform: rotate(180deg);
    `};
`;

const SecondaryMenuItem = styled.div`
  padding-right: ${rem(20)};
`;

const StyledIcon = styled.div<{ $size?: number }>`
  && {
    width: ${p => rem(p.$size || 20)};
    height: ${p => rem(p.$size || 20)};
  }
`;

export interface MobileNavbarProps {
  isMobileNavFolded?: boolean;
  isSideFolded?: boolean;
  onMobileNavToggle?: () => void;
  onSideToggle?: () => void;
  showSideNav?: boolean;
}

const MobileNavbar = ({
  children,
  isMobileNavFolded,
  isSideFolded,
  onMobileNavToggle,
  onSideToggle,
  showSideNav,
}: React.PropsWithChildren<MobileNavbarProps>) => {
  return (
    <Wrapper>
      {showSideNav !== false && (
        <NavButton $active={!isSideFolded} onClick={onSideToggle}>
          {isSideFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>
      )}

      <LogoLink>
        <Logo $compact />
      </LogoLink>

      {children}

      <NavButton onClick={onMobileNavToggle} $active={!isMobileNavFolded} style={{ position: 'absolute', right: 0 }}>
        <ArrowWrapper $shouldRotate={!isMobileNavFolded}>
          <StyledIcon as={KeyboardArrowDown} $size={36} />
        </ArrowWrapper>
      </NavButton>

      <SecondaryMenu $isOpen={!isMobileNavFolded}>
        <NavLinks />
        <SecondaryMenuItem>
          <Social />
        </SecondaryMenuItem>
      </SecondaryMenu>
    </Wrapper>
  );
};

export default MobileNavbar;
