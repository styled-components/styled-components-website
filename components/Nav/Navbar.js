import React, { PureComponent, createRef } from 'react';
import styled, { css } from 'styled-components';
import { Close } from '@styled-icons/material';
import rem from '../../utils/rem';
import { blmGrey, paleGrey } from '../../utils/colors';
import { navbarHeight } from '../../utils/sizes';
import { headerFont } from '../../utils/fonts';
import { mobile } from '../../utils/media';
import Link from '../Link';
import NavLinks from './NavLinks';
import Social from './Social';
import Logo from './Logo';
import MobileNavbar from './MobileNavbar';
import SearchWithAlgolia from './SearchWithAlgolia';

const Wrapper = styled.nav`
  position: fixed;
  left: 0;
  box-sizing: border-box;
  z-index: 3;

  width: 100%;
  height: ${rem(navbarHeight)};

  font-family: ${headerFont};
  font-size: ${rem(15)};
  font-weight: 500;
  background: ${(props) => (props.transparent ? 'transparent' : blmGrey)};
  transition: background 300ms ease-out;
  color: white;
  padding: 0;
`;

const StartWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

const EndWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;
/* stylelint-disable */
const StyledSocial = styled(Social)``;
/* stylelint-enable */

const NormalNavbar = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${rem(20)};
  justify-content: space-between;
  ${StartWrapper}, ${EndWrapper} ${StyledSocial} {
    ${mobile(css`
      display: none;
    `)};
  }
`;

const StyledModalCloseIcon = styled(Close)`
  width: ${rem(26)};
  height: ${rem(26)};
  color: white;
`;

const AlgoliaModal = styled.div`
  ${mobile(css`
    background: currentColor;
    overflow: auto;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  `)};
`;

const baseZ = 1;

const AlgoliaModalHeader = styled.div`
  display: none;
  color: currentColor;

  ${mobile(css`
    display: ${(props) => (props.isOpen ? 'block' : 'none')};

    button {
      cursor: pointer;
      padding: 0;
      position: fixed;
      right: ${rem(10)};
      top: ${rem(11)};
      border: none;
      z-index: ${baseZ + 1};
    }
  `)};
`;

const AlgoliaModalOverlay = styled.div`
  margin-right: ${rem(10)};

  ${mobile(css`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: ${baseZ};
    left: 0;
    display: ${(props) => (props.isOpen ? 'block' : 'none')};
    background: ${paleGrey};
    overflow-y: auto;
    margin: 0;

    .algolia-autocomplete .ds-dropdown-menu {
      position: static !important;
      display: block;
      max-width: 100%;
      min-width: 0;
    }
  `)};
`;

class ModalContainer extends PureComponent {
  modalElement = createRef();
  onModalOverlayClick = (e) => {
    if (!this.modalElement.current.contains(e.target)) {
      this.props.requestModalClose();
    }
  };
  onCloseButtonClick = (e) => {
    e.stopPropagation();
    this.props.requestModalClose();
  };
  render() {
    return (
      <>
        <AlgoliaModalHeader isOpen={this.props.isOpen}>
          <button onClick={this.onCloseButtonClick}>
            <StyledModalCloseIcon />
          </button>
        </AlgoliaModalHeader>
        <AlgoliaModalOverlay onClick={this.onModalOverlayClick} isOpen={this.props.isOpen}>
          <AlgoliaModal ref={this.modalElement}>
            <div>{this.props.children}</div>
          </AlgoliaModal>
        </AlgoliaModalOverlay>
      </>
    );
  }
}

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
}))`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`;

class Navbar extends PureComponent {
  state = {
    isOpen: false,
  };
  openModal = () => this.setState(() => ({ isOpen: true }));
  closeModal = () => this.setState(({ isOpen }) => (isOpen ? { isOpen: false } : null));
  render() {
    const { onSideToggle, onMobileNavToggle, isSideFolded, isMobileNavFolded, showSideNav, transparent } = this.props;

    return (
      <Wrapper transparent={transparent}>
        <NormalNavbar>
          <StartWrapper>
            <LogoLink>
              <Logo />
            </LogoLink>
            <NavLinks />
          </StartWrapper>
          <EndWrapper>
            <ModalContainer isOpen={this.state.isOpen} requestModalClose={this.closeModal}>
              <SearchWithAlgolia requestModalClose={this.closeModal} />
            </ModalContainer>
            <StyledSocial />
          </EndWrapper>
        </NormalNavbar>
        <MobileNavbar
          isSideFolded={isSideFolded}
          onSearchButtonClick={this.openModal}
          isMobileNavFolded={isMobileNavFolded}
          onSideToggle={onSideToggle}
          onMobileNavToggle={onMobileNavToggle}
          showSideNav={showSideNav}
        />
      </Wrapper>
    );
  }
}

export default Navbar;
