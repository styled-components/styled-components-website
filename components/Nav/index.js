import React from 'react';
import Navbar from './Navbar';
import Sidebar from './Sidebar';
import { DocsSidebarMenu, SimpleSidebarMenu } from './SidebarMenus';

const Nav = props => {
  const {
    isSideFolded,
    isMobileNavFolded,
    onSideToggle,
    onMobileNavToggle,
    onRouteChange,
    showSideNav,
    useDocsSidebarMenu,
    transparent,
    pages,
  } = props;

  return (
    <div>
      <Navbar
        showSideNav={showSideNav}
        transparent={transparent}
        isSideFolded={isSideFolded}
        isMobileNavFolded={isMobileNavFolded}
        onSideToggle={onSideToggle}
        onMobileNavToggle={onMobileNavToggle}
      />

      {showSideNav !== false && (
        <Sidebar isFolded={isSideFolded}>
          {useDocsSidebarMenu !== false ? (
            <DocsSidebarMenu onRouteChange={onRouteChange} />
          ) : (
            <SimpleSidebarMenu onRouteChange={onRouteChange} pages={pages} />
          )}
        </Sidebar>
      )}
    </div>
  );
};

export default Nav;
