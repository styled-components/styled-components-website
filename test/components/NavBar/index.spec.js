import React from 'react';
import { mount } from 'enzyme';
import Nav from '../../../components/Nav';

test('Nav renders correctly', () => {
  const wrapper = mount(<Nav />);

  expect(wrapper).toMatchSnapshot();
});
