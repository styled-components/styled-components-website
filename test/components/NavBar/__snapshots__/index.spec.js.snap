// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
.c0 {
  position: fixed;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  display: block;
  z-index: 1;
  font-family: "Avenir Next", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  left: 0;
  top: 0;
  bottom: 0;
  right: auto;
  width: 16.666666666666668rem;
  height: 100%;
  background: linear-gradient(20deg, rgb(219, 112, 147), #daa357);
  box-sizing: border-box;
  color: white;
  overflow-y: scroll;
}

.c1 {
  height: 16.666666666666668rem;
  padding: 3.611111111111111rem 4.722222222222222rem 2.5rem;
}

.c2 {
  display: inline-block;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  background-image: url(/static/logo.png);
  background-size: contain;
  background-position: center;
}

.c3 {
  display: none;
  margin-left: 0.8333333333333334rem;
  margin-top: 0.2777777777777778rem;
  font-size: 1rem;
}

.c5 {
  display: inline-block;
  text-decoration: none;
  color: inherit;
  cursor: pointer;
  padding: 0.1111111111111111rem 0.4444444444444444rem;
  margin: -0.1111111111111111rem -0.4444444444444444rem;
}

.c7 {
  display: block;
  box-sizing: border-box;
}

.c8 {
  display: block;
  box-sizing: border-box;
  height: 100%;
}

.c9 {
  margin-bottom: 1.6666666666666667rem;
}

.c10 {
  display: block;
  margin: 0.5555555555555556rem 2.2222222222222223rem;
  font-weight: normal;
}

.c11 {
  display: block;
  margin: 0.5555555555555556rem 2.2222222222222223rem 0.5555555555555556rem 3.0555555555555554rem;
  font-size: 0.9rem;
  font-weight: normal;
}

.c4 {
  display: none;
  height: 100%;
  padding: 1.1111111111111112rem 0;
  margin: 0 0 0 auto;
  color: white;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c4:active {
  opacity: 0.8;
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.c4:hover {
  background: none;
}

.c6 {
  height: 1.6666666666666667rem;
}

@media (max-width: 62.5em) {
  .c0 {
    bottom: auto;
    right: 0;
    height: auto;
    width: 100%;
    overflow-y: visible;
  }
}

@media (max-width: 62.5em) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    height: 3.888888888888889rem;
    padding: 0 0.8333333333333334rem;
  }
}

@media (max-width: 62.5em) {
  .c2 {
    background-image: url(/static/icon.png);
    background-position: center;
    width: 3.888888888888889rem;
    height: 3.888888888888889rem;
  }
}

@media (max-width: 62.5em) {
  .c3 {
    display: block;
  }
}

@media (min-width: 62.5em) {
  .c5 {
    border-radius: 0.16666666666666666rem;
  }

  .c5:hover {
    background: rgba(20, 20, 20, 0.1);
  }
}

@media (max-width: 62.5em) {
  .c7 {
    height: 0;
    -webkit-transition: height 0.2s ease-in-out;
    transition: height 0.2s ease-in-out;
    overflow: hidden;
  }
}

@media (max-width: 62.5em) {
  .c8 {
    border-top: 2px solid rgba(20, 20, 20, 0.1);
    padding-top: 0.5555555555555556rem;
    padding-bottom: 3.3333333333333335rem;
    overflow-y: scroll;
  }
}

@media (max-width: 62.5em) {
  .c4 {
    display: block;
  }
}

<Navbar>
  <styled.nav>
    <styled.nav>
      <nav
        className="c0"
      >
        <styled.div>
          <div
            className="c1"
          >
            <Link
              href="/docs"
              unstyled={true}
            >
              <Link
                href="/docs"
              >
                <a
                  href="/docs"
                  onClick={[Function]}
                >
                  <styled.div>
                    <div
                      className="c2"
                    />
                  </styled.div>
                </a>
              </Link>
            </Link>
            <styled.div>
              <div
                className="c3"
              >
                <Link
                  href="/docs"
                  unstyled={true}
                >
                  <Link
                    href="/docs"
                  >
                    <a
                      href="/docs"
                      onClick={[Function]}
                    >
                      styled-components
                    </a>
                  </Link>
                </Link>
              </div>
            </styled.div>
            <Navbar
              onClick={[Function]}
            >
              <Styled(styled.a)
                onClick={[Function]}
              >
                <styled.a
                  className="c4"
                  onClick={[Function]}
                >
                  <a
                    className="c4 c5"
                    onClick={[Function]}
                  >
                    <Styled(ThreeBarsIcon)>
                      <ThreeBarsIcon
                        className="c6"
                        height={null}
                        width={null}
                      >
                        <svg
                          className="octicons octicons-three-bars-icon c6"
                          height={null}
                          viewBox="0 0 12 16"
                          width={null}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11.41 9H.59C0 9 0 8.59 0 8c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zm0-4H.59C0 5 0 4.59 0 4c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zM.59 11H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1H.59C0 13 0 12.59 0 12c0-.59 0-1 .59-1z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </ThreeBarsIcon>
                    </Styled(ThreeBarsIcon)>
                  </a>
                </styled.a>
              </Styled(styled.a)>
            </Navbar>
          </div>
        </styled.div>
        <Menu
          isFolded={true}
          onRouteChange={[Function]}
        >
          <styled.aside
            isFolded={true}
          >
            <aside
              className="c7"
            >
              <styled.div>
                <div
                  className="c8"
                >
                  <styled.div
                    onClick={[Function]}
                  >
                    <div
                      className="c9"
                      onClick={[Function]}
                    >
                      <styled.h4>
                        <h4
                          className="c10"
                        >
                          <Link
                            href="/docs/basics"
                          >
                            <Link
                              href="/docs/basics"
                            >
                              <styled.a
                                href="/docs/basics"
                                onClick={[Function]}
                              >
                                <a
                                  className="c5"
                                  href="/docs/basics"
                                  onClick={[Function]}
                                >
                                  Basics
                                </a>
                              </styled.a>
                            </Link>
                          </Link>
                        </h4>
                      </styled.h4>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#motivation"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#motivation"
                            >
                              Motivation
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#installation"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#installation"
                            >
                              Installation
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#getting-started"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#getting-started"
                            >
                              Getting Started
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#passed-props"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#passed-props"
                            >
                              Passed props
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#adapting-based-on-props"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#adapting-based-on-props"
                            >
                              Adapting based on props
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#styling-any-components"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#styling-any-components"
                            >
                              Styling any components
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#extending-styles"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#extending-styles"
                            >
                              Extending styles
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#attaching-additional-props"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#attaching-additional-props"
                            >
                              Attaching additional props
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#animations"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#animations"
                            >
                              Animations
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/basics#react-native"
                          >
                            <a
                              className="c5"
                              href="/docs/basics#react-native"
                            >
                              React Native
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                    </div>
                  </styled.div>
                  <styled.div
                    onClick={[Function]}
                  >
                    <div
                      className="c9"
                      onClick={[Function]}
                    >
                      <styled.h4>
                        <h4
                          className="c10"
                        >
                          <Link
                            href="/docs/advanced"
                          >
                            <Link
                              href="/docs/advanced"
                            >
                              <styled.a
                                href="/docs/advanced"
                                onClick={[Function]}
                              >
                                <a
                                  className="c5"
                                  href="/docs/advanced"
                                  onClick={[Function]}
                                >
                                  Advanced
                                </a>
                              </styled.a>
                            </Link>
                          </Link>
                        </h4>
                      </styled.h4>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#theming"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#theming"
                            >
                              Theming
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#refs"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#refs"
                            >
                              Refs
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#security"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#security"
                            >
                              Security
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#existing-css"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#existing-css"
                            >
                              Existing CSS
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#media-templates"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#media-templates"
                            >
                              Media Templates
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#tagged-template-literals"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#tagged-template-literals"
                            >
                              Tagged Template Literals
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#server-side-rendering"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#server-side-rendering"
                            >
                              Server Side Rendering
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/advanced#babel-plugin"
                          >
                            <a
                              className="c5"
                              href="/docs/advanced#babel-plugin"
                            >
                              Babel Plugin
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                    </div>
                  </styled.div>
                  <styled.div
                    onClick={[Function]}
                  >
                    <div
                      className="c9"
                      onClick={[Function]}
                    >
                      <styled.h4>
                        <h4
                          className="c10"
                        >
                          <Link
                            href="/docs/api"
                          >
                            <Link
                              href="/docs/api"
                            >
                              <styled.a
                                href="/docs/api"
                                onClick={[Function]}
                              >
                                <a
                                  className="c5"
                                  href="/docs/api"
                                  onClick={[Function]}
                                >
                                  API Reference
                                </a>
                              </styled.a>
                            </Link>
                          </Link>
                        </h4>
                      </styled.h4>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/api#primary"
                          >
                            <a
                              className="c5"
                              href="/docs/api#primary"
                            >
                              Primary
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/api#helpers"
                          >
                            <a
                              className="c5"
                              href="/docs/api#helpers"
                            >
                              Helpers
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/api#supported-css"
                          >
                            <a
                              className="c5"
                              href="/docs/api#supported-css"
                            >
                              Supported CSS
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/api#flow"
                          >
                            <a
                              className="c5"
                              href="/docs/api#flow"
                            >
                              Flow
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/api#typescript"
                          >
                            <a
                              className="c5"
                              href="/docs/api#typescript"
                            >
                              TypeScript
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                    </div>
                  </styled.div>
                  <styled.div
                    onClick={[Function]}
                  >
                    <div
                      className="c9"
                      onClick={[Function]}
                    >
                      <styled.h4>
                        <h4
                          className="c10"
                        >
                          <Link
                            href="/docs/faqs"
                          >
                            <Link
                              href="/docs/faqs"
                            >
                              <styled.a
                                href="/docs/faqs"
                                onClick={[Function]}
                              >
                                <a
                                  className="c5"
                                  href="/docs/faqs"
                                  onClick={[Function]}
                                >
                                  FAQs
                                </a>
                              </styled.a>
                            </Link>
                          </Link>
                        </h4>
                      </styled.h4>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/faqs#can-i-nest-rules"
                          >
                            <a
                              className="c5"
                              href="/docs/faqs#can-i-nest-rules"
                            >
                              Can I nest rules?
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/faqs#can-i-refer-to-other-components"
                          >
                            <a
                              className="c5"
                              href="/docs/faqs#can-i-refer-to-other-components"
                            >
                              Can I refer to other components?
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                      <styled.h5>
                        <h5
                          className="c11"
                        >
                          <styled.a
                            href="/docs/faqs#can-i-use-css-frameworks"
                          >
                            <a
                              className="c5"
                              href="/docs/faqs#can-i-use-css-frameworks"
                            >
                              Can I use css frameworks?
                            </a>
                          </styled.a>
                        </h5>
                      </styled.h5>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </aside>
          </styled.aside>
        </Menu>
      </nav>
    </styled.nav>
  </styled.nav>
</Navbar>
`;
